packages/flutter/lib/src/widgets/animated_cross_fade.dart

```diff
@@ -0,0 +1,207 @@
+    this.secondCurve: Curves.linear,
+    this.sizeCurve: Curves.linear,
+    @required this.crossFadeState,
+    @required this.duration
+  }) : super(key: key) {
+    assert(this.firstCurve != null);
+    assert(this.secondCurve != null);
+    assert(this.sizeCurve != null);
+  }
+
+  /// The first child. It fades in an out according to [crossFadeState].
+  final Widget firstChild;
+
+  /// The second child. It fades in an out according to [crossFadeState].
+  final Widget secondChild;
+
+  /// The state towards which the widget is animating.
+  final CrossFadeState crossFadeState;
+
+  /// The duration of the whole orchestrated animation.
+  final Duration duration;
+
+  /// The fade curve of the first child. This curve is inverted, i.e. the first
+  /// widget is visible when this curve provides the value 0.0, and is hidden
+  /// when the curve provides the value 1.0.
+  final Curve firstCurve;
+
+  /// The fade curve of the second child.
+  final Curve secondCurve;
+
+  /// The curve of the animation between the two children's sizes.
+  final Curve sizeCurve;
+
+  @override
+  _AnimatedCrossFadeState createState() => new _AnimatedCrossFadeState();
+}
+
+class _AnimatedCrossFadeState extends State<AnimatedCrossFade> {
+  _AnimatedCrossFadeState() : super();
+
+  AnimationController _controller;
+  Animation<double> _firstAnimation;
+  Animation<double> _secondAnimation;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = new AnimationController(duration: config.duration);
+    _firstAnimation = new Tween<double>(
+      begin: 1.0,
+      end: 0.0
+    ).animate(
+      new CurvedAnimation(
+        parent: _controller,
+        curve: config.firstCurve
+      )
+    );
+    _secondAnimation = new CurvedAnimation(
+      parent: _controller,
+      curve: config.secondCurve
+    );
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  void didUpdateConfig(AnimatedCrossFade oldConfig) {
+    super.didUpdateConfig(oldConfig);
+
+    switch (config.crossFadeState) {
+      case CrossFadeState.showFirst:
+        _controller.reverse();
+        break;
+      case CrossFadeState.showSecond:
+        _controller.forward();
+        break;
+    }
+
+    if (config.duration != oldConfig.duration)
+      _controller.duration = config.duration;
+    if (config.firstCurve != oldConfig.firstCurve) {
+      _firstAnimation = new Tween<double>(
+        begin: 1.0,
+        end: 0.0
+      ).animate(
+        new CurvedAnimation(
+          parent: _controller,
+          curve: config.firstCurve
+        )
+      );
+    }
+    if (config.secondCurve != oldConfig.secondCurve) {
+      _secondAnimation = new CurvedAnimation(
+        parent: _controller,
+        curve: config.secondCurve
+      );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Stack stack;
+
+    if (_controller.status == AnimationStatus.completed ||
```

Comment:
You could just define List<Widget> stackChildren here and create the Stack in the return statement.